HiP-UserStore/Core/CacheDatabaseManager.cs

```diff
@@ -46,50 +48,94 @@ private void ApplyEvent(IEvent ev)
+                    _db.GetCollection<User>(ResourceTypes.User.Name).UpdateOne(x => x.Id == e.Id, update3);
                     break;
 
-                case UserPhotoUploaded e:
-                    var update2 = Builders<User>.Update
-                        .Set(x => x.ProfilePicturePath, e.Path)
-                        .Set(x => x.Timestamp, e.Timestamp);
+                case CreatedEvent e:
+                    var resourceType = e.GetEntityType();
+                    switch (resourceType)
+                    {
+                        case ResourceType _ when resourceType == ResourceTypes.User:
+                            var newUser = new User()
+                            {
+                                Id = e.Id,
+                                UserId = e.UserId,
+                                Timestamp = e.Timestamp
+                            };
+                            _db.GetCollection<User>(ResourceTypes.User.Name).InsertOne(newUser);
+                            break;
 
-                    _db.GetCollection<User>(ResourceType.User.Name).UpdateOne(x => x.Id == e.Id, update2);
+                        case ResourceType _ when resourceType == ResourceTypes.StudentDetails:
+                            var user = _db.GetCollection<User>(ResourceTypes.User.Name).AsQueryable().First(x => x.Id == e.Id);
+                            var userArgs = user.CreateUserArgs2();
+                            var updatedUser = new User(userArgs)
+                            {
+                                Id = e.Id,
+                                UserId = e.UserId,
+                                Timestamp = e.Timestamp,
+                                StudentDetails = new StudentDetails()
+                            };
+                            _db.GetCollection<User>(ResourceTypes.User.Name).ReplaceOne(x => x.Id == e.Id, updatedUser);
+                            break;
+                    }
                     break;
 
-                case UserPhotoDeleted e:
-                    var update3 = Builders<User>.Update
-                        .Set(x => x.ProfilePicturePath, null)
-                        .Set(x => x.Timestamp, e.Timestamp);
+                case PropertyChangedEvent e:
+                    resourceType = e.GetEntityType();
+                    switch (resourceType)
+                    {
+                        case ResourceType _ when resourceType == ResourceTypes.User:
+                            var originalUser = _db.GetCollection<User>(ResourceTypes.User.Name).AsQueryable().First(x => x.Id == e.Id);
+                            var userArgs = originalUser.CreateUserArgs2();
+                            e.ApplyTo(userArgs);
+                            var updatedUser = new User(userArgs)
+                            {
+                                Id = e.Id,
+                                UserId = e.UserId,
```

Comment:
The UserId contains the Id of the user that has created the entity. So the entity id should be kept when you create new events. So concrete it should be  `UserId = originalUser.UserId`. The same holds for the other cases for PropertyChanged events